File: typescript-basic/102/tokenize.ts
const expression: string = "3 + 5 * (2 - 8)";
const tokens: string[] = expression.match(/\d+|\+|\-|\*|\/|\(|\)/g) || [];
console.log(tokens); // ['3', '+', '5', '*', '(', '2', '-', '8', ')']

File: c#-basics/103/tokenize.cs
using System;
using System.Text.RegularExpressions;

class Program {
    static void Main() {
        string expression = "3 + 5 * (2 - 8)";
        var tokens = Regex.Matches(expression, @"\d+|\+|\-|\*|\/|\(|\)");
        foreach (var token in tokens) {
            Console.WriteLine(token);
        }
        // Output: 3, +, 5, *, (, 2, -, 8, )
    }
}
